src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -84,48 +84,86 @@ public static void ConnectManagedWithInstanceNameTest(bool useMultiSubnetFailove
+            string hostname = "";
+            string instanceName = "";
+            int port = -1;
+            DataTestUtility.ParseDataSource(builder.DataSource, out hostname, out port, out instanceName);
+            Assert.True(port != -2, "Named pipe protocol in data source does not support port number.");
+            Assert.True(!string.IsNullOrEmpty(hostname), "Hostname must be included in the data source.");
+            Assert.True(!string.IsNullOrEmpty(instanceName), "Instance name must be included in the data source.");
 
-            bool condition = IsBrowserAlive(hostname) && IsValidInstance(hostname, instanceName);
-            Assert.True(condition, "Browser service is not running or instance name is invalid");
+            bool isBrowserRunning = IsBrowserAlive(hostname);
+            Assert.True(isBrowserRunning, "Browser service is not running.");
 
-            if (condition)
+            bool isInstanceExisting = IsValidInstance(hostname, instanceName);
+            Assert.True(isInstanceExisting, "Instance name is invalid.");
+
+            if (isBrowserRunning && isInstanceExisting)
             {
+                // Create a connection object to ensure SPN info is available via reflection
                 using SqlConnection connection = new(builder.ConnectionString);
                 connection.Open();
-                using SqlCommand command = new("SELECT auth_scheme, local_tcp_port from sys.dm_exec_connections where session_id = @@spid", connection);
-                using SqlDataReader reader = command.ExecuteReader();
-                Assert.True(reader.Read(), "Expected to receive one row data");
-                Assert.Equal("KERBEROS", reader.GetString(0));
-                int localTcpPort = reader.GetInt32(1);
-
-                int spnPort = -1;
-                string spnInfo = GetSPNInfo(builder.DataSource, out spnPort);
-
-                // sample output to validate = MSSQLSvc/machine.domain.tld:spnPort"
-                Assert.Contains($"MSSQLSvc/{hostname}", spnInfo);
-                // the local_tcp_port should be the same as the inferred SPN port from instance name
-                Assert.Equal(localTcpPort, spnPort);
+
+                // Get the SPN info using reflection
+                string spnInfo = GetSPNInfo(builder.DataSource);
+
+                // The expected output to validate is supposed to be in the format "MSSQLSvc/machine.domain.tld:spnPort".
+                // So, we want to get the port number from the SPN and ensure it is a valid port number.
+                string[] spnStrs = spnInfo.Split(':');
+                int portInSPN = 0;
+                if (spnStrs.Length > 1)
+                {
+                    int.TryParse(spnStrs[1], out portInSPN);
+                }
+                Assert.True(portInSPN > 0, "The expected SPN must include a valid port number.");
+
+                connection.Close();
             }
```

Comment:
This doesn't look like it follows some of the code convention I've seen.
Add { } for large code blocks please.

using (sqlConnection) <- this is ok or for one statement deep to use implicit using, and no {} braces>
using (sqlCommand)
{
    A bunch of stuff.....
    more stuff....
    more stuff....
}

The lack of brackets IMO, makes this harder to read. And even if you are implying { }, connection.Close(); should be before the Assert. 
Anycode after an exception is Asserted is not executed. 
Because of the implicit using { } , the connection.Close() is irrelevant.

Reply:
The missused using has been removed.